bookkeeper-server/src/main/java/org/apache/bookkeeper/client/ReadOnlyLedgerHandle.java

```diff
@@ -178,4 +180,19 @@ public String toString() {
     protected void initializeExplicitLacFlushPolicy() {
         explicitLacFlushPolicy = ExplicitLacFlushPolicy.VOID_EXPLICITLAC_FLUSH_POLICY;
     }
+
+    @Override
+    public void asyncReadLastEntry(ReadCallback cb, Object ctx) {
+        asyncReadLastConfirmed(new ReadLastConfirmedCallback() {
+            @Override
+            public void readLastConfirmedComplete(int rc, long lastConfirmed, Object ctx) {
+                if (lastConfirmed < 0) {
```

Comment:
you need check `rc` first. if `rc` is not `OK`, then fail `ReadCallback`. Otherwise do the logic to read actual entry.